### PR TITLE
Remove infinite loop on attempts to rebuild

### PR DESCRIPTION
**Pull request by dvoytenko**

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -344,30 +344,37 @@ export class Resources {
...
     }
   }
 
   /**
    * Builds resources that are ready to be built.
+   * @param {boolean=} scheduleWhenBuilt
    * @private
    */
-  buildReadyResources_() {
+  buildReadyResources_(scheduleWhenBuilt = true) {
```
Comment (mkhatib) [inline src/service/resources-impl.js]: You forgot to pass this to `buildReadyResourcesUnsafe_` in this function.

File: test/functional/test-resources.js [hunk 2]
```diff
@@ -850,6 +850,32 @@ describe('Resources discoverWork', () => {
...
+    const schedulePassStub = sandbox.stub(resources, 'schedulePass');
+    sandbox.stub(resources, 'schedule_');
+    resources.documentReady_ = true;
+    resource1.state_ = ResourceState.NOT_BUILT;
+    resource1.blacklisted_ = true;
+    resource1.build = sandbox.spy();
+
+    resources.discoverWork_();
+
+    expect(resource1.build.callCount).to.equal(0);
```
Comment (mkhatib) [inline test/functional/test-resources.js]: Prefer `to.not.be.called` for better error messages when test fails

File: test/functional/test-resources.js [hunk 3]
```diff
@@ -850,6 +850,32 @@ describe('Resources discoverWork', () => {
...
+  it('should build resource when not built', () => {
+    const schedulePassStub = sandbox.stub(resources, 'schedulePass');
+    sandbox.stub(resources, 'schedule_');
+    resources.documentReady_ = true;
+    resource1.state_ = ResourceState.NOT_BUILT;
+    resource1.build = sandbox.spy();
+
+    resources.discoverWork_();
+
+    expect(resource1.build.callCount).to.equal(1);
```
Comment (mkhatib) [inline test/functional/test-resources.js]: ditto: `to.be.calledOnce`

File: src/service/resources-impl.js (hunk 1 above)
Comment (dvoytenko) [inline src/service/resources-impl.js]: Fixed. Added more tests for this scenario.

File: test/functional/test-resources.js (hunk 3 above)
Comment (dvoytenko) [inline test/functional/test-resources.js]: Done

File: test/functional/test-resources.js (hunk 2 above)
Comment (dvoytenko) [inline test/functional/test-resources.js]: Done.

**Comment by dvoytenko** [comment]
@mkhatib All addressed. PTAL.

**Comment by erwinmombay** [comment]
@dvoytenko do we need this in this weeks canary? Im cutting a new one again today

**Comment by dvoytenko** [comment]
@erwinmombay If you are cutting canary today, then yes, please.

**Comment by mkhatib** [comment]
LGTM

**Pull request merged**